src/routes/load/container/Load.tsx

```diff
@@ -16,17 +16,25 @@ import { SAFELIST_ADDRESS } from 'src/routes/routes'
 import { buildSafe } from 'src/routes/safe/store/actions/fetchSafe'
 import { history } from 'src/store'
 import { loadFromStorage } from 'src/utils/storage'
+import { Dispatch } from 'redux'
+import { SafeOwner, SafeRecordProps } from '../../safe/store/models/safe'
+import { List } from 'immutable'
 
-export const loadSafe = async (safeName, safeAddress, owners, addSafe) => {
+export const loadSafe = async (
+  safeName: string,
+  safeAddress: string,
+  owners: List<SafeOwner>,
+  addSafe: Dispatch<any>,
+): Promise<void> => {
   const safeProps = await buildSafe(safeAddress, safeName)
   safeProps.owners = owners
 
-  await addSafe(safeProps)
+  const storedSafes: SafeRecordProps = (await loadFromStorage(SAFES_KEY)) || {}
```

Comment:
type is incorrect

Reply:
why?

Reply:
You load a list of safes, not an individual safe object

Reply:
Also I’m not sure if the safes coming from loadFromStorage are using the correct data structures SafeRecordProps contains (e.g Set)

Reply:
please check again